### PR TITLE
feat(client): add copy content button to text file preview modal

### PR DESCRIPTION
**Pull request by maraf**
## Overview

This PR adds a copy-to-clipboard button to the text file preview modal, allowing users to quickly copy file content without needing to download the file first.

### Feature Implementation
- Added `filePreviewTextCopied` reactive state to track copy feedback
- Implemented `handleCopyTextPreview()` method using the Clipboard API
- Copy button (clipboard icon → green checkmark) appears only when `filePreviewType === 'text'`
- Visual feedback: Icon changes to green checkmark for 1.5s after successful copy

### Layout Fix
Fixed a CSS scrollbar gutter issue that created unwanted padding on the right side of the modal:
- Changed outer modal container from `overflow-hidden` to `overflow-clip`
- `overflow-clip` provides identical visual clipping without creating a CSS scroll container
- On Windows classic scrollbars, `overflow-hidden` reserves ~17px for scrollbar gutter; `overflow-clip` does not
- Reorganized nested divs to properly separate scrollable region from padded content

## Visual Verification

Updated all 68 Playwright screenshots across desktop (1280×800) and mobile (390×844) viewports to reflect:
- Copy button rendering on text file previews
- Resolved right-side spacing gap in modal layout
- No regressions on other preview types (image, video, audio, PDF)

### CODE REVIEW

File: src/SharedSpaces.Client/src/features/space-view/space-view.ts [hunk 1]
```diff
@@ -3211,14 +3226,28 @@ export class SpaceView extends BaseElement {
...
             </h3>
             <div class="flex shrink-0 items-center gap-2">
+              ${this.filePreviewType === 'text' && this.filePreviewText != null
+                ? html`
+                    <button
+                      @click=${this.handleCopyTextPreview}
+                      class="rounded p-1 text-slate-400 transition hover:text-white"
+                      title=${this.filePreviewTextCopied ? 'Copied!' : 'Copy content'}
+                      aria-label=${this.filePreviewTextCopied ? 'Copied to clipboard' : 'Copy file content to clipboard'}
+                    >
```
[the remark this answers is not in the document]
Comment (maraf) [reply src/SharedSpaces.Client/src/features/space-view/space-view.ts:3257]: Applied in ff7086e. Added coverage in SpaceView - File Preview Modal for: (1) copy button renders only for text previews, and (2) clicking copy calls navigator.clipboard.writeText and switches title/aria-label to the copied state, then resets after advancing fake timers by 1500ms. Also added tests for empty-string copy and latest-click timeout behavior.

File: src/SharedSpaces.Client/src/features/space-view/space-view.ts [hunk 2]
```diff
@@ -1933,6 +1934,20 @@ export class SpaceView extends BaseElement {
...
     this.filePreviewLoading = false;
     this.filePreviewError = '';
+    this.filePreviewTextCopied = false;
+  };
+
+  private handleCopyTextPreview = async () => {
+    if (!this.filePreviewText) return;
+    try {
+      await navigator.clipboard.writeText(this.filePreviewText);
+      this.filePreviewTextCopied = true;
```
[the remark this answers is not in the document]
Comment (maraf) [reply src/SharedSpaces.Client/src/features/space-view/space-view.ts:1953]: Applied in ff7086e. handleCopyTextPreview now uses a null-or-undefined guard so empty text is still copyable, and the copied-state timeout is tracked and cleared so older clicks cannot reset newer feedback early. The timeout is also cleared in closeFilePreview and in component disconnect cleanup.
